src/components/UserMenu.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <v-layout justify-end class="js-user-menu">
+    <div v-if="!loggedIn && !loginFormVisible" class="flex-none">
+      <v-btn flat @click="loginFormVisible = true">Login</v-btn>
+      <v-btn flat @click="$router.push({ name: 'register' })">Register</v-btn>
+    </div>
+    <v-layout v-if="!loggedIn && loginFormVisible" row class="flex-none">
+      <v-text-field v-model="username" label="username" class="login-input"/>
+      <v-text-field v-model="password" label="password" type="password" class="login-input"/>
+      <v-btn flat @click="login">Login</v-btn>
+      <v-btn flat @click="loginFormVisible = false;">Cancel</v-btn>
+    </v-layout>
+    <v-layout v-if="loggedIn" row align-center class="flex-none">
+      <span>Welcome {{ user }},&nbsp;</span>
+      <a href="#" @click.prevent="logout">Logout</a>
+    </v-layout>
+  </v-layout>
+</template>
+<script>
+export default {
+  data() {
+    return {
+      loginFormVisible: false,
+      username: undefined,
+      password: undefined,
+    };
+  },
+  computed: {
```

Comment:
```
import { mapGetters } from 'vuex';
```

```
  ...mapGetters([
    'loggedIn',
    'username',
  ])
```

Reply:
Also you should consider grouping `this.username` and `this.password` into a `this.form` object (or add a prefix) so you don't need to rename store's `username` to `user`.

Reply:
Done